### PR TITLE
[FLINK-16418][hive] Hide hive version to avoid user confuse

### PR DESCRIPTION
**Pull request by JingsongLi**
## What is the purpose of the change

Version in Yaml/HiveCatalog needs to be consistent with the dependencies version. There are three places: version in metastore, version in dependencies, version in Yaml/HiveCatalog, users are easy to make mistakes.

## Brief change log

- Add constructor without hive version to HiveCatalog and HiveModule
- Remove hive version in document

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)

### CODE REVIEW

File: flink-table/flink-sql-client/src/test/resources/test-sql-client-catalogs.yaml [hunk 1]
```diff
@@ -155,7 +155,6 @@ catalogs:
   - name: hivecatalog
     type: hive
     test: test
-    hive-version: 2.3.4
```
Comment (lirui-apache) [inline flink-table/flink-sql-client/src/test/resources/test-sql-client-catalogs.yaml]: I don't think this one should be removed. It's there to verify the version provided by user takes precedence over the version detected in classpath.

**Comment by JingsongLi** [comment]
> LGTM, except that we need documentation stating that
> 
> 1. hive versions will be automatically inferred at runtime
> 2. hive versions in these configs are optional, users still can specify explicitly if they want to

Hi @bowenli86 , I prefer not provide this message, I can not find the case users need to configure.

**Comment by bowenli86** [comment]
what's the "hive version" of Hive in CDH or HDP?

**Comment by JingsongLi** [comment]
> what's the "hive version" of Hive in CDH or HDP?

Is there any users use CDH or HDP client? The users that use CDH I know they just use open-source corresponding hive client dependencies.
Like Spark&Presto, they just use built-in hive client dependencies.

**Comment by bowenli86** [comment]
@lirui-apache can you confirm?

otherwise, I'm +1 for this PR

**Comment by lirui-apache** [comment]
@bowenli86 @JingsongLi For HDP and CDH users, I think they probably should use the HDP and CDH version Hive jars. For example, HDP Hive-1.2.1 has cherry picked lots of patches from newer version and therefore is different from Apache Hive-1.2.1. So using the HDP version usually means users can have the extra bug fixes than the Apache version.

**Comment by JingsongLi** [comment]
> @bowenli86 @JingsongLi For HDP and CDH users, I think they probably should use the HDP and CDH version Hive jars. For example, HDP Hive-1.2.1 has cherry picked lots of patches from newer version and therefore is different from Apache Hive-1.2.1. So using the HDP version usually means users can have the extra bug fixes than the Apache version.

OK, I think this cases should be document, because it is not so easy to do, so I created JIRA: https://issues.apache.org/jira/browse/FLINK-16434 for this.

**Comment by JingsongLi** [comment]
Merging...

**Pull request merged**